llama-index-integrations/node_parser/relational/llama-index-node-parser-relational-dashscope/llama_index/node_parser/relational/dashscope/base.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Callable, List, Optional, Dict
+import logging
+import requests
+import os
+import json
+
+from llama_index.core.bridge.pydantic import Field
+from llama_index.core.node_parser.relational.base_element import (
+    BaseElementNodeParser,
+    Element,
+)
+from llama_index.core.schema import BaseNode, TextNode
+
+
+class DashScopeJsonNodeParser(BaseElementNodeParser):
+    """DashScope Json format element node parser.
+
+    Splits a json format document from DashScope Parse into Text Nodes and Index Nodes
+    corresponding to embedded objects (e.g. tables).
+    """
+
+    try_count_limit: int = Field(
+        default=10, description="Maximum number of retry attempts."
+    )
+    chunk_size: int = Field(default=500, description="Size of each chunk to process.")
+    overlap_size: int = Field(
+        default=100, description="Overlap size between consecutive chunks."
+    )
+    separator: str = Field(
+        default=" |,|，|。|？|！|\n|\\?|\\!",
+        description="Separator characters for splitting texts.",
+    )
+    pip: bool = Field(default=False, description="Flag to enable or disable PIP.")
+    input_type: str = Field(default="idp", description="parse format type.")
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "DashScopeJsonNodeParser"
+
+    def get_nodes_from_node(self, node: TextNode) -> List[BaseNode]:
+        """Get nodes from node."""
+        ftype = node.metadata.get("parse_fmt_type", self.input_type)
+        assert ftype in [
+            "DASHCOPE_DOCMIND",
+            "idp",
+        ], f"Unexpected parse_fmt_type: {node.metadata.get('parse_fmt_type', '')}"
+
+        ftype_map = {
+            "DASHCOPE_DOCMIND": "idp",
+        }
+
+        my_input = {
+            "text": node.get_content(),
+            "file_type": ftype_map.get(ftype, ftype),
+            "chunk_size": self.chunk_size,
+            "overlap_size": self.overlap_size,
+            "language": "cn",
```

Comment:
language is hardcoded to cn?

Reply:
added language to configurable parameters.